### PR TITLE
Fix dialyzer warnings

### PR DESCRIPTION
**Pull request by ananthakumaran**

### CODE REVIEW

File: lib/mongo_ecto.ex [hunk 1]
```diff
@@ -850,7 +848,7 @@ defmodule Mongo.Ecto do
     query = %ReadQuery{coll: "system.namespaces", query: @list_collections_query}
     opts = Keyword.put(opts, :log, false)
 
-    Connection.read(repo, query, opts)
```
Comment (scottmessinger) [inline lib/mongo_ecto.ex:853]: Thanks! Could you tell me more about this change? Is this also dialyzer related? The rest of the changes look straightforward. Thanks!

Comment (ananthakumaran) [reply lib/mongo_ecto.ex:853]: Yes, dialyzer produced a warning for this call. The read/3 function expects adapter_meta as first argument, since this code only gets executed for mongo < 3, this was not tested by CI. I ran the test suite with mongo 2.8 locally and fixed the arguments passed to this call. More details can be found in the [commit](https://github.com/elixir-mongo/mongodb_ecto/pull/185/commits/5c8eedf92a47a3dd87d8277dbfce91ae605b764f) message

Comment (scottmessinger) [reply lib/mongo_ecto.ex:853]: Thanks! I should have looked at the the commit message. Thanks for making it so detailed!

**Comment by ananthakumaran** [comment]
Btw, mix test fails on master as well. I just focused on fixing dialzyer warnings in the PR.

**Comment by scottmessinger** [comment]
Thanks, @ananthakumaran!!!! I really appreciate it!
